src/main/java/io/split/android/client/SyncConfig.java

```diff
@@ -0,0 +1,73 @@
+package io.split.android.client;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.split.android.client.utils.Logger;
+import io.split.android.client.validators.SplitValidator;
+import io.split.android.client.validators.SplitValidatorImpl;
+
+class SyncConfig {
+
+    private final static int MAX_BY_NAME_VALUES = 400;
+    private final static int MAX_BY_PREFIX_VALUES = 50;
+    private final List<SplitFilter> mFilters;
+
+
+    private SyncConfig(List<SplitFilter> filters) {
+        mFilters = filters;
+    }
+
+    public List<SplitFilter> getFilters() {
+        return mFilters;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private List<SplitFilter> mBuilderFilters = new ArrayList<>();
+        private final SplitValidator mSplitValidator = new SplitValidatorImpl();
+
+        public SyncConfig build() {
+            Map<SplitFilter.Type, List<String>> groupedFilters = new HashMap<>();
+            for (SplitFilter filter : mBuilderFilters) {
+                List<String> groupedValues = groupedFilters.get(filter.getType());
+                if (groupedValues == null) {
+                    groupedValues = new ArrayList<>();
+                }
+
+                List<String> values = filter.getValues();
+                for (String value : values) {
+                    if (mSplitValidator.validateName(value) != null) {
+                        Logger.w(String.format("Warning: Malformed %s value. Filter ignored: %s", filter.getType().toString(), value));
+                    } else {
+                        groupedValues.add(value);
+                    }
+                }
+                groupedFilters.put(filter.getType(), groupedValues);
```

Comment:
Dont't forget to move this line inside the if statement ;)

Reply:
Great. Done Doc!